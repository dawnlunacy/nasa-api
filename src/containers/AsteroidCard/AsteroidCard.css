.asteroid-card {
  background-color: black;
  margin: 10px;
  text-align: center;
}