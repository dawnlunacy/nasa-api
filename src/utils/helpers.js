export const findTodaysDate = () =>{
  let today = new Date();
  let dd = String(today.getDate()).padStart(2, '0');
  let mm = String(today.getMonth() + 1).padStart(2, '0');
  let yyyy = today.getFullYear();
  let todayFormat = yyyy + '/' + mm + '/' + dd;
  return todayFormat
}

export const formatTodaysDate = (stringDate) => {
  let formatDate = new Date(stringDate).toString().split(' ').slice(1, 4).join(' ')
  return formatDate
}

export const findDay = () => {
  let today = findTodaysDate();
  let weekday = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
  let day = weekday[new Date(today).getDay()]
  return day
}