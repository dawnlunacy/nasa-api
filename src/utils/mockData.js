export const mockApod = {
  "backgroundImage": "url(https://apod.nasa.gov/apod/image/1911/M8_Wu_966.jpg)",
  "backgroundRepeat": "no-repeat",
  "backgroundSize": "100% 100%"
}
export const mockNextWeekUrl = "http://www.neowsapp.com/rest/v1/feed?start_date=2019-11-07&end_date=2019-11-14&detailed=false&api_key=m98g3WmabopZXIZRCQ0HdHYrEwuHimuH8b8JjicA";

export const mockPrevWeekUrl = "http://www.neowsapp.com/rest/v1/feed?start_date=2019-10-24&end_date=2019-10-31&detailed=false&api_key=m98g3WmabopZXIZRCQ0HdHYrEwuHimuH8b8JjicA";

export const mockUnfilteredApodResponse = {
  "copyright": "Zhuoqun Wu",
  "date": "2019-11-04",
  "explanation": "Stars are battling gas and dust in the Lagoon Nebula but the photographers are winning. Also known as M8, this photogenic nebula is visible even without binoculars towards the constellation of the Archer (Sagittarius). The energetic processes of star formation create not only the colors but the chaos. The glowing gas results from high-energy starlight striking interstellar hydrogen gas and trace amounts of sulfur, and oxygen gases. The dark dust filaments that lace M8 were created in the atmospheres of cool giant stars and in the debris from supernovae explosions. The light from M8 we see today left about 5,000 years ago. Light takes about 50 years to cross this section of M8.",
  "hdurl": "https://apod.nasa.gov/apod/image/1911/M8_Wu_966.jpg",
  "media_type": "image",
  "service_version": "v1",
  "title": "Near the Center of the Lagoon Nebula",
  "url": "https://apod.nasa.gov/apod/image/1911/M8_Wu_966.jpg"
}

export const mockCleanNeos = {
  "2019-11-07": [
    {
      "id": "3879721",
      "name": "(2019 UL6)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879721",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "86.41",
      "estimatedDiameterMax": "193.21",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "17480.89",
          "missEarthDistance": "10538462.96"
        }
      ]
    },
    {
      "id": "3879139",
      "name": "(2019 TM6)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879139",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "691.74",
      "estimatedDiameterMax": "1546.77",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "53573.59",
          "missEarthDistance": "37761662.77"
        }
      ]
    },
    {
      "id": "3879532",
      "name": "(2019 UL4)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879532",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "74.43",
      "estimatedDiameterMax": "166.43",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "20227.06",
          "missEarthDistance": "3976280.61"
        }
      ]
    },
    {
      "id": "3127406",
      "name": "(2002 LT38)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3127406",
      "isPotentiallyHazardous": "true",
      "estimatedDiameterMin": "759.52",
      "estimatedDiameterMax": "1698.34",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "32796.79",
          "missEarthDistance": "37796010.02"
        }
      ]
    },
    {
      "id": "3596685",
      "name": "(2012 BO77)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3596685",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "316.62",
      "estimatedDiameterMax": "707.99",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "27294.30",
          "missEarthDistance": "29643901.45"
        }
      ]
    },
    {
      "id": "3610028",
      "name": "(2012 TE53)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3610028",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "43.71",
      "estimatedDiameterMax": "97.73",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "46451.99",
          "missEarthDistance": "32585498.26"
        }
      ]
    },
    {
      "id": "2467336",
      "name": "467336 (2002 LT38)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2467336",
      "isPotentiallyHazardous": "true",
      "estimatedDiameterMin": "692.69",
      "estimatedDiameterMax": "1548.91",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "32796.75",
          "missEarthDistance": "37795980.94"
        }
      ]
    },
    {
      "id": "3883091",
      "name": "(2019 UO7)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3883091",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "25.01",
      "estimatedDiameterMax": "55.93",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "9168.67",
          "missEarthDistance": "915455.91"
        }
      ]
    },
    {
      "id": "3884022",
      "name": "(2019 UJ9)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3884022",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "103.98",
      "estimatedDiameterMax": "232.50",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-07",
          "relativeVelocity": "14657.60",
          "missEarthDistance": "8352868.98"
        }
      ]
    }
  ],
  "2019-11-06": [
    {
      "id": "3892163",
      "name": "(2019 UM13)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3892163",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "41.07",
      "estimatedDiameterMax": "91.84",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "13259.32",
          "missEarthDistance": "2747841.77"
        }
      ]
    },
    {
      "id": "3884011",
      "name": "(2019 UV8)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3884011",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "184.39",
      "estimatedDiameterMax": "412.31",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "11487.40",
          "missEarthDistance": "23412662.46"
        }
      ]
    },
    {
      "id": "3789636",
      "name": "(2017 WB14)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3789636",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "87.20",
      "estimatedDiameterMax": "195.00",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "25602.86",
          "missEarthDistance": "24398790.78"
        }
      ]
    },
    {
      "id": "2463282",
      "name": "463282 (2012 HR15)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2463282",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "3315.43",
      "estimatedDiameterMax": "7413.53",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "56409.12",
          "missEarthDistance": "39030368.00"
        }
      ]
    },
    {
      "id": "3836110",
      "name": "(2018 VA5)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3836110",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "45.77",
      "estimatedDiameterMax": "102.34",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "34939.41",
          "missEarthDistance": "44109163.53"
        }
      ]
    },
    {
      "id": "3261402",
      "name": "(2004 VA1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3261402",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "692.69",
      "estimatedDiameterMax": "1548.91",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "62437.04",
          "missEarthDistance": "36693638.22"
        }
      ]
    },
    {
      "id": "3616816",
      "name": "(2012 VN7)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3616816",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "1149.58",
      "estimatedDiameterMax": "2570.54",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "4395.80",
          "missEarthDistance": "42438032.94"
        }
      ]
    },
    {
      "id": "3583386",
      "name": "(2011 UC115)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3583386",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "288.76",
      "estimatedDiameterMax": "645.69",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "21346.02",
          "missEarthDistance": "30087907.34"
        }
      ]
    },
    {
      "id": "3172333",
      "name": "(2004 BW18)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3172333",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "275.77",
      "estimatedDiameterMax": "616.63",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "12880.88",
          "missEarthDistance": "41635258.18"
        }
      ]
    },
    {
      "id": "3774014",
      "name": "(2017 HG1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3774014",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "27.58",
      "estimatedDiameterMax": "61.66",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "35460.84",
          "missEarthDistance": "39992232.25"
        }
      ]
    },
    {
      "id": "2481025",
      "name": "481025 (2004 VA1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2481025",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "692.69",
      "estimatedDiameterMax": "1548.91",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "62437.03",
          "missEarthDistance": "36693608.80"
        }
      ]
    },
    {
      "id": "3572706",
      "name": "(2011 OB)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3572706",
      "isPotentiallyHazardous": "true",
      "estimatedDiameterMin": "1149.58",
      "estimatedDiameterMax": "2570.54",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "50511.58",
          "missEarthDistance": "39896277.79"
        }
      ]
    },
    {
      "id": "3667121",
      "name": "(2014 GC35)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3667121",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "288.76",
      "estimatedDiameterMax": "645.69",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "51230.71",
          "missEarthDistance": "29823322.12"
        }
      ]
    },
    {
      "id": "3763429",
      "name": "(2016 VU2)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3763429",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "347.17",
      "estimatedDiameterMax": "776.29",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "37382.37",
          "missEarthDistance": "38906325.89"
        }
      ]
    },
    {
      "id": "3879316",
      "name": "(2019 UT3)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879316",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "72.67",
      "estimatedDiameterMax": "162.49",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "8270.58",
          "missEarthDistance": "7632180.65"
        }
      ]
    },
    {
      "id": "3892010",
      "name": "(2019 UE13)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3892010",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "64.88",
      "estimatedDiameterMax": "145.09",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-06",
          "relativeVelocity": "19826.18",
          "missEarthDistance": "1964242.41"
        }
      ]
    }
  ],
  "2019-11-09": [
    {
      "id": "3780740",
      "name": "(2017 QW1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3780740",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "50.18",
      "estimatedDiameterMax": "112.21",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-09",
          "relativeVelocity": "7522.28",
          "missEarthDistance": "11971127.83"
        }
      ]
    },
    {
      "id": "3490930",
      "name": "(2010 AG79)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3490930",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "913.15",
      "estimatedDiameterMax": "2041.86",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-09",
          "relativeVelocity": "45133.29",
          "missEarthDistance": "33146103.25"
        }
      ]
    }
  ],
  "2019-11-08": [
    {
      "id": "3888751",
      "name": "(2019 UM12)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3888751",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "102.74",
      "estimatedDiameterMax": "229.73",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "30369.23",
          "missEarthDistance": "304492.40"
        }
      ]
    },
    {
      "id": "3348265",
      "name": "(2006 TD1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3348265",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "550.22",
      "estimatedDiameterMax": "1230.34",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "3882.21",
          "missEarthDistance": "26323041.24"
        }
      ]
    },
    {
      "id": "3747991",
      "name": "(2016 FX2)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3747991",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "158.69",
      "estimatedDiameterMax": "354.83",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "11470.27",
          "missEarthDistance": "25406763.31"
        }
      ]
    },
    {
      "id": "2085236",
      "name": "85236 (1993 KH)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2085236",
      "isPotentiallyHazardous": "true",
      "estimatedDiameterMin": "1661.65",
      "estimatedDiameterMax": "3715.57",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "19966.82",
          "missEarthDistance": "9423214.64"
        }
      ]
    },
    {
      "id": "3395956",
      "name": "(2007 WM3)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3395956",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "209.19",
      "estimatedDiameterMax": "467.76",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "36870.14",
          "missEarthDistance": "14309370.67"
        }
      ]
    },
    {
      "id": "3733933",
      "name": "(2015 VH65)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3733933",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "25.15",
      "estimatedDiameterMax": "56.24",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "42374.13",
          "missEarthDistance": "37717979.50"
        }
      ]
    },
    {
      "id": "3879292",
      "name": "(2019 UN2)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879292",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "84.71",
      "estimatedDiameterMax": "189.42",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "12048.75",
          "missEarthDistance": "5645122.79"
        }
      ]
    },
    {
      "id": "3879537",
      "name": "(2019 UR4)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879537",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "37.11",
      "estimatedDiameterMax": "82.99",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-08",
          "relativeVelocity": "10140.38",
          "missEarthDistance": "1748244.11"
        }
      ]
    }
  ],
  "2019-11-03": [
    {
      "id": "3879309",
      "name": "(2019 UH3)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879309",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "58.90",
      "estimatedDiameterMax": "131.71",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "19094.09",
          "missEarthDistance": "2463681.90"
        }
      ]
    },
    {
      "id": "3843082",
      "name": "(2019 MF3)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3843082",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "232.13",
      "estimatedDiameterMax": "519.07",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "22999.40",
          "missEarthDistance": "19383853.89"
        }
      ]
    },
    {
      "id": "3648767",
      "name": "(2013 TH4)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3648767",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "199.77",
      "estimatedDiameterMax": "446.71",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "45682.16",
          "missEarthDistance": "39448456.56"
        }
      ]
    },
    {
      "id": "3719183",
      "name": "(2015 JD1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3719183",
      "isPotentiallyHazardous": "true",
      "estimatedDiameterMin": "661.52",
      "estimatedDiameterMax": "1479.19",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "26682.93",
          "missEarthDistance": "3078707.86"
        }
      ]
    },
    {
      "id": "3856707",
      "name": "(2019 RN4)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3856707",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "977.10",
      "estimatedDiameterMax": "2184.87",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "27767.37",
          "missEarthDistance": "38659168.82"
        }
      ]
    },
    {
      "id": "3878570",
      "name": "(2019 TZ1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3878570",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "128.10",
      "estimatedDiameterMax": "286.43",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "10139.05",
          "missEarthDistance": "8944752.03"
        }
      ]
    },
    {
      "id": "3883108",
      "name": "(2019 UL8)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3883108",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "47.27",
      "estimatedDiameterMax": "105.69",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-03",
          "relativeVelocity": "28606.28",
          "missEarthDistance": "668680.23"
        }
      ]
    }
  ],
  "2019-11-05": [
    {
      "id": "3842660",
      "name": "(2019 KK4)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3842660",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "233.21",
      "estimatedDiameterMax": "521.46",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "19979.99",
          "missEarthDistance": "36174130.82"
        }
      ]
    },
    {
      "id": "3694779",
      "name": "(2014 UG115)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3694779",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "41.74",
      "estimatedDiameterMax": "93.33",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "28868.12",
          "missEarthDistance": "25831063.60"
        }
      ]
    },
    {
      "id": "3607371",
      "name": "(2012 PW)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3607371",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "1515.44",
      "estimatedDiameterMax": "3388.64",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "35085.52",
          "missEarthDistance": "7241451.15"
        }
      ]
    },
    {
      "id": "3645651",
      "name": "(2013 PA39)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3645651",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "55.02",
      "estimatedDiameterMax": "123.03",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "27704.95",
          "missEarthDistance": "37058584.20"
        }
      ]
    },
    {
      "id": "3514891",
      "name": "(2010 GB6)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3514891",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "331.54",
      "estimatedDiameterMax": "741.35",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "52858.42",
          "missEarthDistance": "37184945.80"
        }
      ]
    },
    {
      "id": "3879290",
      "name": "(2019 UL2)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879290",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "89.98",
      "estimatedDiameterMax": "201.20",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "15910.81",
          "missEarthDistance": "4946543.69"
        }
      ]
    },
    {
      "id": "3879544",
      "name": "(2019 UH5)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879544",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "30.67",
      "estimatedDiameterMax": "68.58",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-05",
          "relativeVelocity": "7857.61",
          "missEarthDistance": "3174641.91"
        }
      ]
    }
  ],
  "2019-11-04": [
    {
      "id": "3883084",
      "name": "(2019 UJ7)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3883084",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "100.68",
      "estimatedDiameterMax": "225.13",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "11625.59",
          "missEarthDistance": "14198443.79"
        }
      ]
    },
    {
      "id": "2011885",
      "name": "11885 Summanus (1990 SS)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2011885",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "1739.96",
      "estimatedDiameterMax": "3890.68",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "39502.37",
          "missEarthDistance": "32262758.67"
        }
      ]
    },
    {
      "id": "3702671",
      "name": "(2014 YN)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3702671",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "63.17",
      "estimatedDiameterMax": "141.26",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "26606.33",
          "missEarthDistance": "25227855.41"
        }
      ]
    },
    {
      "id": "3729834",
      "name": "(2015 TX144)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3729834",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "25.15",
      "estimatedDiameterMax": "56.24",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "37377.45",
          "missEarthDistance": "45383464.36"
        }
      ]
    },
    {
      "id": "3748362",
      "name": "(2016 FC12)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3748362",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "104.84",
      "estimatedDiameterMax": "234.44",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "16233.50",
          "missEarthDistance": "32111090.57"
        }
      ]
    },
    {
      "id": "3753692",
      "name": "(2016 LO1)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3753692",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "75.95",
      "estimatedDiameterMax": "169.83",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "30352.63",
          "missEarthDistance": "27834881.02"
        }
      ]
    },
    {
      "id": "3892193",
      "name": "(2019 VD)",
      "nasaUrl": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3892193",
      "isPotentiallyHazardous": "false",
      "estimatedDiameterMin": "28.40",
      "estimatedDiameterMax": "63.51",
      "closeApproachData": [
        {
          "closeApproachDate": "2019-11-04",
          "relativeVelocity": "26839.07",
          "missEarthDistance": "108207.26"
        }
      ]
    }
  ]
}

