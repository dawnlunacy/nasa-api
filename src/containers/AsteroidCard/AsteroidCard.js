import React from 'react';
import './AsteroidCard.css';
import { connect } from 'react-redux';

export const AsteroidCard = ({id, name, nasaUrl, isPotentiallyHazardous, estimatedDiameterMin, estimatedDiameterMax, closeApproachDate, relativeVelocity, missEarthDistance}) => {
 console.log("why", typeof isPotentiallyHazardous)
  
  return (
    <section className="asteroid-card">
      <p> ID: {id} </p>
      <p> Name: {name} </p>
      <p> Close Approach Date: {closeApproachDate} </p> 
      <p> Is Potentially Hazardous: {isPotentiallyHazardous} </p>
      <p> Estimated Diameter Min: {estimatedDiameterMin} Feet </p>
      <p> Estimated Diameter Max: {estimatedDiameterMax} Feet </p>
      <p> Relative Velocity: {relativeVelocity} Miles Per Hour</p>
      <p> Will Miss Earth by: {missEarthDistance} Miles </p>
      <p> Link to Nasa: {nasaUrl} </p>
    </section>
    
  )
}

export const mapStateToProps = state => ({
  neos: state.neos,
  currentNeoDate: state.currentNeoDate
});

export default connect(mapStateToProps, null)(AsteroidCard)