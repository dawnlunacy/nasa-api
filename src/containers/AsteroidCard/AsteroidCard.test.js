import React from 'react';
import { shallow } from 'enzyme';
import { AsteroidCard, mapStateToProps} from '../AsteroidCard/AsteroidCard';
import { mockCleanNeo } from '../../utils/mockData';

describe('AsteroidCard', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = shallow(<AsteroidCard
      neos={ mockCleanNeo }
      currentNeoDate='2019-11-03'
      id={"3879721"}
      name={"(2019 UL6)"}
      nasaUrl={"http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879721"}
      isPotentiallyHazardous={false}
      estimatedDiameterMin={86.4052294554}
      estimatedDiamerMax={193.2079666738}
      closeApproachDate={"2019-11-07"}
      relativeVelocity={"17480.8945933894"}
      missEarthDistance={"10538462.9640548218"}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  
});