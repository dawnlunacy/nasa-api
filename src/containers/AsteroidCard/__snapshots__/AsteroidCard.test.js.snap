// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsteroidCard should match the snapshot 1`] = `
<section
  className="asteroid-card"
>
  <p
    className="p-ac"
  >
     ID: 
    3879721
     
  </p>
  <p
    className="p-ac"
  >
     Name: 
    (2019 UL6)
     
  </p>
  <p
    className="p-ac"
  >
     Close Approach Date: 
    2019-11-07
     
  </p>
  <p
    className="p-ac"
  >
     Is Potentially Hazardous: 
     
  </p>
  <p
    className="p-ac"
  >
     Estimated Diameter Min: 
    86.4052294554
     Feet 
  </p>
  <p
    className="p-ac"
  >
     Estimated Diameter Max: 
     Feet 
  </p>
  <p
    className="p-ac"
  >
     Relative Velocity: 
    17480.8945933894
     Miles Per Hour
  </p>
  <p
    className="p-ac"
  >
     Will Miss Earth by: 
    10538462.9640548218
     Miles 
  </p>
  <a
    className="p-ac-link"
    href="http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3879721"
  >
     View in NASA database  
  </a>
</section>
`;
